### PR TITLE
Add Capitalization doc guidelines to Community section

### PR DESCRIPTION
**Pull request by chenette**
This commit includes a new "Doc Guidelines" section with all planned guidelines (to minimize merge conflicts). Some links will be broken until the other guideline PRs have been merged.

Note: This file contains an internal TOC for increased usability (for those readers who don't notice or use right-nav menus).

Signed-off-by: Anne Chenette <chenette@bitwise.io>

### CODE REVIEW

File: community/doc_guidelines/capitalization.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+* ID, IDs
+
+## J
+
+* JavaScript
+* JSON
+
+## K
+
+* key registry
```
Comment (agunde406) [inline community/doc_guidelines/capitalization.md:72]: I think key registry can be removed

Comment (agunde406) [reply community/doc_guidelines/capitalization.md:72]: or marked as obsolete

File: community/doc_guidelines/capitalization.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+## O
+
+* OK
+* orchestrator
+
+## P
+
+* PDF _(not "pdf" or ".pdf")_
+* Pike smart contract
+* Private Counter
```
Comment (agunde406) [inline community/doc_guidelines/capitalization.md:98]: Private Counter and Private XO examples were removed, maybe mark as obsolete

File: community/doc_guidelines/capitalization.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+* XO smart contract
+* `xo` Sawtooth command _(always lowercase)_
+
+## Y
+
+* YAML file, YAML format (not "yaml file" or ".yaml file")
+
+## Z
+
+* ZeroMQ, ZMQ, 0MQ, ØMQ _(To do: Identify the preferred term)_
```
Comment (agunde406) [inline community/doc_guidelines/capitalization.md:152]: Is the todo intentionally left in here?

File: community/doc_guidelines/capitalization.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+* For titles and headings, follow the Chicago Manual of Style's title rules.
+  (Tip: Use [capitalizemytitle.com](https://capitalizemytitle.com/) and select
+  "Chicago".)
+
+    - Capitalize the first and the last word, nouns, pronouns, adjectives,
+      verbs, adverbs, and subordinate conjunctions (such as Because, Once,
+      Therefore, When, and Unless).
+    - Lowercase articles (a, an, the), coordinating conjunctions, prepositions,
+      and the "to" in infinitive verbs.
+
```
[the remark this answers is not in the document]
Comment (chenette) [reply community/doc_guidelines/capitalization.md:32]: Fixed.

File: community/doc_guidelines/capitalization.md (hunk 1 above)
Comment (chenette) [reply community/doc_guidelines/capitalization.md]: Marked this (and Private Counter & XO terms) as obsolete for now.  A later PR is planned to add new terms and remove the obsolete ones.

File: community/doc_guidelines/capitalization.md (hunk 3 above)
Comment (chenette) [reply community/doc_guidelines/capitalization.md]: Yes, but it's probably better not to expose internal TO DO items.  Deleted.

**Pull request merged**